uwblueprint/community-fridge-kw
# feat: get and delete checkin endpoints

## Brief description. What is this change? 

### [Get and Delete CheckIn endpoint](https://www.notion.so/uwblueprintexecs/Get-and-Delete-CheckIn-endpoint-473b5e391b304e49bdc7f2b8c81ed52a)



## Implementation description. How did you make this change?
* API endpoints for the get (GET) and delete (DELETE) were created, and corresponding services `getAllCheckIns(), getCheckInsById(id: string), getCheckInsByVolunteerId(volunteerId: string), deleteCheckInById(id: string), deleteCheckInsByDateRange(startDate: string, endDate: string)`. 



## Steps to test
Get functions:
1. Insert data to PgAdmin under ```checkIn``` table (right click on checkIn table -> scripts -> INSERT Script)
2. For getting all: ```localhost:5000/checkIn``` and ensure it gets the right response
3. For getting by id: ```localhost:5000/checkIn/<id>```
4. For getting by volunteerId: ```localhost:5000/checkIn?volunteerId=<volunteerId>```
5. Make sure the get all, get by id, get by volunteer id returns the correct responses.

Delete functions:
1. insert data like above
2. For deleting by id, ```localhost:5000/checkin/<id>``` and ensures it gets deleted from table
3. For deleting by date range  ```localhost:5000//checkin?startDate=<startDate>?endDate=<endDate>``` -> make sure the parameters are correct in Postman before running the request 
^ example:  ```localhost:5000//checkin?startDate=2022-01-31T05:00:00.000Z?endDate=2022-02-31T05:00:00.000Z```

## Checklist
